src/core/control/xojfile/LoadHandler.cpp

```diff
@@ -394,528 +248,365 @@ void LoadHandler::parseBgPixmap() {
+                          const fs::path& filename, size_t timestamp) {
+    xoj_assert(!this->text);
+    this->text = std::make_unique<Text>();
 
+    XojFont& f = this->text->getFont();
+    f.setName(font);
+    f.setSize(size);
     this->text->setX(x);
     this->text->setY(y);
+    this->text->setColor(color);
 
-    XojFont& f = text->getFont();
-    f.setName(sFont);
-    f.setSize(fontSize);
-    const char* sColor = LoadHandlerHelper::getAttrib("color", false, this);
-    Color color{0U};
-    LoadHandlerHelper::parseColor(sColor, color, this);
-    text->setColor(color);
+    setAudioAttributes(*this->text, filename, timestamp);
+}
 
-    const char* fn = LoadHandlerHelper::getAttrib("fn", true, this);
-    if (fn != nullptr && strlen(fn) > 0) {
-        if (this->isGzFile) {
-            text->setAudioFilename(fn);
-        } else {
-            auto tempFile = getTempFileForPath(fn);
-            if (!tempFile.empty()) {
-                text->setAudioFilename(tempFile.string());
-            }
-        }
-    }
+void LoadHandler::setTextContents(const std::string& contents) {
+    xoj_assert(this->text);
 
-    size_t ts = 0;
-    if (LoadHandlerHelper::getAttribSizeT("ts", true, this, ts)) {
-        text->setTimestamp(ts);
-    }
+    this->text->setText(contents);
+}
+
+void LoadHandler::finalizeText() {
+    xoj_assert(this->text);
+
+    this->layer->addElement(std::move(this->text));
 }
 
-void LoadHandler::parseImage() {
-    double left = LoadHandlerHelper::getAttribDouble("left", this);
-    double top = LoadHandlerHelper::getAttribDouble("top", this);
-    double right = LoadHandlerHelper::getAttribDouble("right", this);
-    double bottom = LoadHandlerHelper::getAttribDouble("bottom", this);
+void LoadHandler::addImage(double left, double top, double right, double bottom) {
+    xoj_assert(!this->image);
+    this->image = std::make_unique<Image>();
 
-    xoj_assert(this->image == nullptr);
-    auto imageOwn = std::make_unique<Image>();
-    this->image = imageOwn.get();
-    this->layer->addElement(std::move(imageOwn));
     this->image->setX(left);
     this->image->setY(top);
     this->image->setWidth(right - left);
     this->image->setHeight(bottom - top);
 }
 
-void LoadHandler::parseTexImage() {
-    double left = LoadHandlerHelper::getAttribDouble("left", this);
-    double top = LoadHandlerHelper::getAttribDouble("top", this);
-    double right = LoadHandlerHelper::getAttribDouble("right", this);
-    double bottom = LoadHandlerHelper::getAttribDouble("bottom", this);
-
-    const char* imText = LoadHandlerHelper::getAttrib("text", false, this);
-    const char* compatibilityTest = LoadHandlerHelper::getAttrib("texlength", true, this);
-    auto imTextLen = strlen(imText);
-    if (compatibilityTest != nullptr) {
-        imTextLen = LoadHandlerHelper::getAttribSizeT("texlength", this);
+void LoadHandler::setImageData(std::string&& data) {
+    xoj_assert(this->image);
```

Comment:
Why? Declaring it as an rvalue reference saves one move operation. I was actually thinking about doing this for many other calls as well. 
`std::forward` on the line below should be changed back to a plain `std::move` though.

Reply:
There are several points to consider:

A move operation for a string is literally free (copy Pointer to reg, store 0 at old position, optimize subsequent moves away, store pointer to target). 

Also std::move has no effect on the assembly, it just marks a reference as move from on assign/construction.

Using T&& forces a user to be explicit, whether an object may be moved or copied. That may be desirable (e.g. a vector of tremendous size, which should never be copied.).

The drawback is, it forces the user to be explicit, increases verbosity and may be annoying when you nearly always want to do a copy anyway.

When you std::move into such a function there will no move happen, but the caller assumes this, since it has called std::move. The rvalue-ref is still a reference, you can do stuff with it, and when you don't move it away, it still exists at the original position. Leading to at least unexpected behav., e.g. deferred deletion. 

A move may be faster than a reference (caching, locality.) - weak argument

In the end, it is a tradeoff. 
Is the extra verbosity worthy? Or does it annoy others?
I and the average C++ dev would say, that it is not worth it in this case.

Reply:
So whenever I need to save a movable object, I should use move (or an implicit copy at function call),
and whenever a function is read-only, I can keep a const (lvalue) ref. 

Does that sound about correct?